uni/lib/unittest.icn

```diff
@@ -0,0 +1,166 @@
+# TODO: assertions
+#     - asserting objects
+#     - asserting nested values (list, table, objects, etc.)
+# TODO: Test Reporter
+#     - add message to results
+#     - add backtrace if possible
+# TODO: exit 0 when success, 1 when failure.
+# TODO: there should be a way to discover and load tests dynamically
+# TODO: add timer, see tests/unicon/tester.icn
+# TODO: how to catch stdin/stdout/stderr within tests, see capture().
+# TODO: enable/disable verbosity.
+
+package unittest
+
+link ximage
+
+import lang
+
+invocable all
+
+class TestCase : Object(__status)
+   $define STATUS_SUCCESS 0
+   $define STATUS_FAIL 1
+
+   method setupClass()
+      write(classname(self), "::setupClass()")
+   end
+
+   method teardownClass()
+      write(classname(self), "::teardownClass()")
+   end
+
+   method setup()
+      write(classname(self), "::setup()")
+   end
+
+   method teardown()
+      write(classname(self), "::teardown()")
+   end
+
+   method __reset()
+      self.__status := STATUS_SUCCESS
+   end
+
+   method assert(expression)
+      if not @expression[1] then {
+         self.__status := STATUS_FAIL
+         write("Assertion failed.")
+         fail
+      }
+
+      return
+   end
+
+   method assertEqual(expected, actual)
+      if expected ~== actual then {
+         write("Assertion failed: expected ", expected, ", but got ", actual)
+         self.__status := STATUS_FAIL
+         fail
+      }
+
+      return
+   end
+
+initially
+   /__status := STATUS_SUCCESS
+end
+
+class TestSuite(__tests, __testReporter)
+   method add(testCase)
+      cname := classname(testCase)
+      if (\cname & (not (cname == "unittest__TestCase"))) then {
```

Comment:
slightly shorter? `if classname(testCase) ~=== "unittest__TestCase" then...`

Reply:
- [ ] Update how to ensure the class added is not directly an instance of TestCase.